src/main/utils/mavenUtils.ts

```diff
@@ -0,0 +1,319 @@
+        const artifactId: RegExpMatchArray | null = parentText[0].toString().match(/(?<=\<artifactId>).+?(?=\<\/artifactId>)/i);
+        const version: RegExpMatchArray | null = parentText[0].toString().match(/(?<=\<version>).+?(?=\<\/version>)/i);
+        return groupId + ':' + artifactId + ':' + version;
+    }
+    return '';
+}
+
+export function deepSearchNode(array: PomTree[], pomId: string): PomTree | undefined {
+    for (let index: number = 0; index < array.length; index++) {
+        if (pomId === array[index].pomId) {
+            return array[index];
+        }
+        if (!!array[index].children) {
+            deepSearchNode(array[index].children!, pomId);
+        }
+    }
+    return;
+}
+
+export function searchPomId(array: PomTree[], pomId: string): number {
+    for (let index: number = 0; index < array.length; index++) {
+        if (pomId === array[index].pomId) {
+            return index;
+        }
+    }
+    return -1;
+}
+
+/**
+ *  @example
+ * --- maven-dependency-plugin:2.8:tree (default-cli) @ multi2 ---
+ *      org.jfrog.test:multi2:jar:3.7-SNAPSHOT
+ *     \- javax.servlet.jsp:jsp-api:jar:2.1:compile
+ * @returns 'org.jfrog.test:multi2:3.7-SNAPSHOT'
+ * @param rawDependency Raw dependency text
+ */
+export function getProjectInfo(rawDependency: string): [string, string, string] {
+    return getDependencyInfo(rawDependency.replace(/\s/g, '') + ':dummyText');
+}
+
+/**
+ *@example of raw dependency "|  |  +- javax.mail:mail:jar:1.4:compile"
+ * @param rawDependency
+ * @returns [groupId,ArtifactId,verion] aka [javax.mail, mail, 1.4]
+ */
+export function getDependencyInfo(rawDependency: string): [string, string, string] {
+    let result: string[] = rawDependency.split(':');
+    //skip none alphabet letters
+    let startIndex: number = result[0].search(/[A-Za-z0-9]/);
+    return [result[0].slice(startIndex), result[1], result[result.length - 2]];
+}
+export function searchPomDependencies(pomId: string, dependenciesList: string[]): string | undefined {
+    for (const iterator of dependenciesList) {
+        const [groupId, artifactId, version] = getProjectInfo(iterator);
+        if (groupId + ':' + artifactId + ':' + version === pomId) {
+            return iterator;
+        }
+    }
+    return;
+}
+
+// 'mvn dependency:tree' duplicate the parent dependencies to its child.
+// this method filter out parent dependencies from child dependency
+export function FilterParentDependencies(parentDeps: DependenciesTreeNode[], childNode: PomTree) {
+    parentDeps.forEach(element => {
+        if (!(element instanceof MavenTreeNode)) {
+            const regex: RegExp = new RegExp(`^.*${element.label}.*$`, 'mg');
+            childNode.rawDependencies = childNode.rawDependencies.replace(regex, '').trim();
+        }
+    });
+}
+
+function executeMavenCmd(mvnCommand: string, pomPath: string): void {
+    exec.execSync(mvnCommand, { cwd: path.dirname(pomPath) });
```

Comment:
Following https://github.com/jfrog/jfrog-vscode-extension/pull/22, please add `maxBuffer` after merging from master.
```suggestion
    exec.execSync(mvnCommand, { cwd: path.dirname(pomPath), maxBuffer: this.SPAWN_PROCESS_BUFFER_SIZE });
```